pennylane/beta/tapes/qnode.py

```diff
@@ -392,11 +392,35 @@ def construct(self, args, kwargs):
             )
 
     def __call__(self, *args, **kwargs):
+
+        if self.interface == "autograd":
+            # HOTFIX: to maintain compatibility with core, here we treat
+            # all inputs that do not explicitly specify `requires_grad=False`
+            # as trainable. This should be removed at some point, forcing users
+            # to specify `requires_grad=True` for trainable parameters.
+            args = [
+                anp.array(a, requires_grad=True) if not hasattr(a, "requires_grad") else a
+                for a in args
+            ]
+
         # construct the tape
         self.construct(args, kwargs)
 
         # execute the tape
-        return self.qtape.execute(device=self.device)
+        res = self.qtape.execute(device=self.device)
+
+        # HOTFIX: to maintain compatibility with core, we squeeze
+        # all outputs.
+
+        # Get the namespace associated with the return type
+        res_type_namespace = res.__class__.__module__.split(".")[0]
+
+        if res_type_namespace in ("pennylane", "autograd"):
+            # For PennyLane and autograd we must branch, since
+            # 'squeeze' does not exist in the top-level of the namespace
+            return anp.squeeze(res)
+
+        return __import__(res_type_namespace).squeeze(res)

```

Comment:
I wonder if this could have any performance implications (e.g. reporting a module on each call?). Is importing a heavy operation?

Reply:
I couldn't see any noticeable difference when testing the autograd and torch interfaces, so all good!

Reply:
> I wonder if this could have any performance implications (e.g. importing a module on each call?). Is importing a heavy operation?

I don't believe so. When Python imports a library, it remains in memory and is never re-imported on subsequent calls within the same session. It's why we can have `import tensorflow` in multiple PennyLane modules, and it never causes performance issues!

Reply:
And in this case, if the tape is returning a autograd/torch/tf datatype, it means that autograd/torch/tf have _already_ been imported somewhere earlier in the pipeline :)

Reply:
Thanks, that's great! Explains why the performance times remained the same